library/common/extensions/key_value/platform/config.cc

```diff
@@ -63,14 +64,16 @@ PlatformKeyValueStoreFactory::createStore(const Protobuf::Message& config,
   const auto& typed_config = MessageUtil::downcastAndValidate<
       const ::envoy::config::common::key_value::v3::KeyValueStoreConfig&>(config,
                                                                           validation_visitor);
-  const auto file_config = MessageUtil::anyConvertAndValidate<
+  const auto platform_kv_store_config = MessageUtil::anyConvertAndValidate<
       envoymobile::extensions::key_value::platform::PlatformKeyValueStoreConfig>(
       typed_config.config().typed_config(), validation_visitor);
-  auto milliseconds =
-      std::chrono::milliseconds(DurationUtil::durationToMilliseconds(file_config.save_interval()));
-  return std::make_unique<PlatformKeyValueStore>(dispatcher, milliseconds,
-                                                 platform_interface_.value().get(),
-                                                 file_config.max_entries(), file_config.key());
+  auto milliseconds = std::chrono::milliseconds(
+      DurationUtil::durationToMilliseconds(platform_kv_store_config.save_interval()));
+  std::unique_ptr<PlatformInterface> platform_interface =
+      std::make_unique<PlatformInterfaceImpl>(platform_kv_store_config.kv_store_name());
```

Comment:
I think I need Mike to look at this, but it looks like this gets passed to 
*static_cast<envoy_kv_store*>(Api::External::retrieveApi(name))
and there's a second key used for read and write
so my mental model of this (which could be 100% wrong) would be that we should be calling registerAPI early on in the engine as we do for the event tracker, and we'd register it with a global well known name, and then the individual proto config keys would be passed into the read/write calls.

Reply:
If we use a single global well-known name, then we can only use the PlatformKeyValueStore for one purpose, not multiple purposes.  Enabling the registered KV store to be specified in the PlatformKeyValueStoreConfig allows the PlatformKeyValueStore to be used for multiple purposes.

builder.addKeyValueStore("dns_cache", new SharedPreferencesStore())
builder.addKeyValueStore("xds_cache", new SharedPreferencesStore())

Then for DNS cache, the PlatformKeyValueStoreConfig would have `kv_store_name` set to `dns_cache` and for the xDS cache, the PlatformKeyValueStoreConfig would have `kv_store_name` set to `xds_cache`.

Reply:
"If we use a single global well-known name, then we can only use the PlatformKeyValueStore for one purpose, not multiple purposes. "
Why?  I thought the name here was a handle on how one gets the KV store platform, and then later on we'd 
platform.write("dns_cache", dns_cache_contents)
platform.read("dns_cache")
platform.read("xds_cache")

Reply:
Alyssa and I chatted over VC, I changed the code to use a single globally-known name for the KV store.